tests/test_random_number_generator.h

```diff
@@ -106,6 +106,26 @@ TEST_CASE("[RandomNumberGenerator] Restore from seed") {
 	CHECK_MESSAGE(s0_1_before == s0_1_after, msg);
 	CHECK_MESSAGE(s0_2_before == s0_2_after, msg);
 }
+
+TEST_CASE("[RandomNumberGenerator] Random int range bias check") {
```

Comment:
Not sure about this, but may be worth allowing this to fail without affecting CI, similarly to #44560.

```suggestion
TEST_CASE_MAY_FAIL("[RandomNumberGenerator] Random int range bias check") {
```